File: java/src/main/java/vsdk/toolkit/render/sgl/Polygon.java
package vsdk.toolkit.render.sgl;

// Note: don't put > 32 doubles in Poly_vert, or mask will overflow
public class Polygon {
    public int n; // Number of sides
    public long mask; // Interpolation mask for vertex elems
    public PolygonVertex[] vertices;

    public Polygon() {
        vertices = new PolygonVertex[PolygonClipResultInfo.MAXIMUM_SIDES_PER_POLYGON];
        for (int i = 0; i < vertices.length; i++) {
            vertices[i] = new PolygonVertex();
        }
    }
}

File: typescript/base/src/main/vsdk/toolkit/render/sgl/Polygon.ts
import { PolygonClipResultInfo } from "./PolygonClipResultInfo";
import { PolygonVertex } from "./PolygonVertex";

export class Polygon {
  public n: number;
  public mask: number;
  public vertices: PolygonVertex[];

  public constructor() {
    this.n = 0;
    this.mask = 0;
    this.vertices = new Array<PolygonVertex>(PolygonClipResultInfo.MAXIMUM_SIDES_PER_POLYGON);
    for (let i = 0; i < this.vertices.length; i++) {
      this.vertices[i] = new PolygonVertex();
    }
  }
}
